### PR TITLE
Allow Display and Oreo to be disabled to save flash

### PR DESCRIPTION
**Pull request by andyp1per**
This PR allows Display and Oreo to be disabled and disables them on MatekF405-Wing and re-enables RunCam and Spektrum telemetry. The flash size stays the same.

Other candidates would be:

- Mount (there is already a feature)
- Robotis
- Volz
- Beacon
- Largely unused backends for different drivers

A huge source of flash cost is range finder. Do we really need all these options (and the number of allowed rangefinders) on by default?

### CODE REVIEW

**Comment by WickedShell** [comment]
OreoLED could probably be opt in only, and only add it to the solo builds in the hwdef since it doesn't really make sense anywhere else.... But this would also work.

**Comment by Hwurzburg** [comment]
fyi, I use mount on many FPV planes to control the pan and get servo stretching....would not like that one gone...

**Comment by khancyr** [comment]
Without proximity we could probably reduce the number of rangefinder to 4
Robotis driver may not be highly used so it could but put aside

File: libraries/AP_Notify/OreoLED_I2C.h [hunk 1]
```diff
@@ -19,6 +19,10 @@
 #include <AP_HAL/AP_HAL.h>
 #include "NotifyDevice.h"
 
+#ifndef HAL_WITH_OREO_LED
```
Comment (peterbarker) [inline libraries/AP_Notify/OreoLED_I2C.h:22]: I think our pattern would be
```suggestion
#ifndef HAL_OREO_LED_ENABLED
```

ATM.  Similarly elsewhere.  97 of the former, 52 of the latter in some bodgy grepping.  Closer than I thought.

File: libraries/AP_HAL_ChibiOS/hwdef/MatekF405-Wing/hwdef.dat [hunk 2]
```diff
@@ -199,6 +199,7 @@ define GENERATOR_ENABLED 0
 define AC_OAPATHPLANNER_ENABLED 0
 define PRECISION_LANDING 0
 define HAL_BARO_WIND_COMP_ENABLED 0
-define HAL_RUNCAM_ENABLED 0
```
Comment (peterbarker) [inline libraries/AP_HAL_ChibiOS/hwdef/MatekF405-Wing/hwdef.dat:202]: This should be a separate PR IMO.  They should come in on their merits rather than because other things have been disabled in the same PR.  I'd approve that PR, BTW.

File: libraries/AP_HAL_ChibiOS/hwdef/MatekF405-Wing/hwdef.dat [hunk 3]
```diff
@@ -199,6 +199,7 @@ define GENERATOR_ENABLED 0
...
-define HAL_SPEKTRUM_TELEM_ENABLED 0
 define HAL_SOARING_ENABLED 0
+define HAL_WITH_DISPLAY 0
+define HAL_WITH_OREO_LED 0
+define GRIPPER_ENABLED 0
```
Comment (peterbarker) [inline libraries/AP_HAL_ChibiOS/hwdef/MatekF405-Wing/hwdef.dat:205]: This was a bit of a shocker to be missed.

File: libraries/AP_HAL_ChibiOS/hwdef/MatekF405-Wing/hwdef.dat [hunk 4]
```diff
@@ -199,6 +199,7 @@ define GENERATOR_ENABLED 0
 define AC_OAPATHPLANNER_ENABLED 0
 define PRECISION_LANDING 0
 define HAL_BARO_WIND_COMP_ENABLED 0
-define HAL_RUNCAM_ENABLED 0
-define HAL_SPEKTRUM_TELEM_ENABLED 0
 define HAL_SOARING_ENABLED 0
+define HAL_WITH_DISPLAY 0
+define HAL_WITH_OREO_LED 0
```
Comment (peterbarker) [inline libraries/AP_HAL_ChibiOS/hwdef/MatekF405-Wing/hwdef.dat:204]: I agree with @WickedShell this should be off by default and enabled by the Solo builds.

**Review by peterbarker — CHANGES_REQUESTED** [review]

File: libraries/AP_Notify/OreoLED_I2C.h [hunk 5]
```diff
@@ -19,6 +19,10 @@
...
+#define HAL_WITH_OREO_LED !HAL_MINIMIZE_FEATURES
```
Comment (tridge) [inline libraries/AP_Notify/OreoLED_I2C.h:23]: add a flash size 1024 check for that too I think

**Review by tridge — CHANGES_REQUESTED** [review]
thanks for this!

File: libraries/AP_Notify/OreoLED_I2C.h (hunk 1 above)
Comment (andyp1per) [reply libraries/AP_Notify/OreoLED_I2C.h]: Ok I have gone with ENABLED

File: libraries/AP_HAL_ChibiOS/hwdef/MatekF405-Wing/hwdef.dat (hunk 2 above)
Comment (andyp1per) [reply libraries/AP_HAL_ChibiOS/hwdef/MatekF405-Wing/hwdef.dat:202]: In my defence I am literally copying what tridge did to disable these!

File: libraries/AP_HAL_ChibiOS/hwdef/MatekF405-Wing/hwdef.dat (hunk 4 above)
Comment (andyp1per) [reply libraries/AP_HAL_ChibiOS/hwdef/MatekF405-Wing/hwdef.dat]: Done

File: libraries/AP_Notify/OreoLED_I2C.h (hunk 5 above)
Comment (andyp1per) [reply libraries/AP_Notify/OreoLED_I2C.h]: I actually just turned it off as per above. I put the flash check in for display though

**Review by tridge — APPROVED** [review]

**Comment by tridge** [comment]
needs rebase

**Comment by peterbarker** [comment]
Merged, thanks!
